keras/layers/local.py

```diff
@@ -0,0 +1,408 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from keras import backend as K
+from keras.layers import activations, initializations, regularizers, constraints
+from keras.engine import Layer, InputSpec
+
+
+def conv_output_length(input_length, filter_size, border_mode, stride):
+    if input_length is None:
+        return None
+    assert border_mode in {'same', 'valid'}
+    if border_mode == 'same':
+        output_length = input_length
+    elif border_mode == 'valid':
+        output_length = input_length - filter_size + 1
+    return (output_length + stride - 1) // stride
+
+
+class LocallyConnected1D(Layer):
+    # TODO batch_dot
+    '''LocallyConnected1D layer works almost the same as Convolution1D layer,
+    except that weights are unshared. When using this layer as the first layer
```

Comment:
Maybe some precision about what this means is needed, e.g. "a different weight matrix is used for each different patch of the input"...